cmd/terramate/cli/cli.go

```diff
@@ -1127,6 +1148,108 @@ func (c *cli) checkGenCode() bool {
 	return true
 }
 
+func (c *cli) eval() {
+	logger := log.With().
+		Str("action", "cli.Eval()").
+		Logger()
+
+	ctx := c.setupEvalContext()
+	for _, exprStr := range c.parsedArgs.Experimental.Eval.Exprs {
+		expr, err := eval.ParseExpressionBytes([]byte(exprStr))
+		if err != nil {
+			logger.Fatal().Err(err).Send()
+		}
+
+		val, err := ctx.Eval(expr)
+		if err != nil {
+			logger.Fatal().Err(err).
+				Str("expr", exprStr).
+				Msg("evaluating expression")
+		}
+
+		var out []byte
+		if c.parsedArgs.Experimental.Eval.AsJSON {
+			out, err = json.Marshal(val, val.Type())
+			if err != nil {
+				logger.Fatal().
+					Str("expr", exprStr).
+					Err(err).
+					Msgf("converting value %s to json", val.GoString())
+			}
+		} else {
+			tokens, err := eval.TokensForValue(val)
+			if err != nil {
+				logger.Fatal().
+					Str("expr", exprStr).
+					Err(err).
+					Msgf("serializing value %s", val.GoString())
```

Comment:
```suggestion
					Msgf("tokenizing value %s", val.GoString())
```

Reply:
tokenizing: string -> tokens
serializing: object -> string

Reply:
hmm even though what you are doing is trying to produce a string in the end, what just failed is exactly the process of get the tokens (val -> tokens), so it seemed to me like a tokenization error :thinking: (in order to serialize you were generating tokens and that failed).

Reply:
the user doesn't need to know that... what failed was the process of producing a string out of a runtime hcl value.

Reply:
Then it is more subtle, your answer implied that I didn't even know what serialization is, but the code reporting the error is called right after a function call that is returning tokens (doing tokenization).